src/store_rebuild.cc

```diff
@@ -109,19 +109,52 @@ storeCleanup(void *)
                    "cache index (at your own risk).\n", store_errors);
         }
 
-        if (store_digest)
-            storeDigestNoteStoreReady();
-
         currentSearch = NULL;
+        Store::Root().markValidated();
+        RunRegisteredHere(RegisteredRunner::useFullyIndexedStore);
     } else
         eventAdd("storeCleanup", storeCleanup, NULL, 0.0, 1);
 }
 
+static void
+StoreRebuildFinalize()
+{
+    assert(Store::Disks::AllIndexed());
+
+    /*
+     * We are done reading or scanning all cache_dirs. Now report the stats and start
+     * the validation (storeCleanup()) thread.
+     */
+
+    if (Store::Disks::Active()) {
```

Comment:
The `stats.scancount` field is also not an ideal candidate because it might, at least in theory, be zero after indexing was completed (e.g. because the UFS-like cache dir had no cache entries and does not use pre-allocated slots for future entries). In this hypothetical case, reporting those zeros is meaningful.

Can we use `counts.started()` instead?